crates/pint-query/src/main.rs

```diff
@@ -0,0 +1,153 @@
+use clap::{builder::styling::Style, Parser};
+use essential_rest_client::node_client::EssentialNodeClient;
+use essential_types::{convert::word_from_bytes, ContentAddress};
+use pint_abi::types::{ContractABI, TypeABI};
+use pint_manifest::ManifestFile;
+use std::{fs::read_dir, path::PathBuf, str::FromStr};
+
+#[derive(Parser, Debug)]
+#[command(version, about, long_about = None)]
+/// Tool to query state from an Essential node endpoint.
+struct Args {
+    /// The endpoint of node to bind to.
+    #[arg(long)]
+    node_address: String,
+    /// The contract address to query, encoded as hex.
+    #[arg(long)]
+    contract_address: ContentAddress,
+    /// The path to the package manifest.
+    ///
+    /// If not provided, the current directory is checked and then each parent
+    /// recursively until a manifest is found.
+    #[arg(long)]
+    manifest_path: Option<PathBuf>,
+    /// The key to query, encoded as hex.
```

Comment:
Perhaps we should have something like:

```suggestion
    /// The key to query, encoded as hex.
    ///
    /// Note that only one of either `key` OR `<STORAGE_ACCESS>` may be provided.
```